src/paho/mqtt/client.py

```diff
@@ -3494,25 +3806,29 @@ def _handle_pubrec(self):
                 msg.timestamp = time_func()
                 return self._send_pubrel(mid)
 
-        return MQTT_ERR_SUCCESS
+        return MQTTErrorCode.MQTT_ERR_SUCCESS
 
-    def _handle_unsuback(self):
+    def _handle_unsuback(self) -> MQTTErrorCode:
         if self._protocol == MQTTv5:
             if self._in_packet['remaining_length'] < 4:
-                return MQTT_ERR_PROTOCOL
+                return MQTTErrorCode.MQTT_ERR_PROTOCOL
         elif self._in_packet['remaining_length'] != 2:
-            return MQTT_ERR_PROTOCOL
+            return MQTTErrorCode.MQTT_ERR_PROTOCOL
 
         mid, = struct.unpack("!H", self._in_packet['packet'][:2])
         if self._protocol == MQTTv5:
             packet = self._in_packet['packet'][2:]
             properties = Properties(UNSUBACK >> 4)
             props, props_len = properties.unpack(packet)
-            reasoncodes = []
+            reasoncodes_list = []
             for c in packet[props_len:]:
-                reasoncodes.append(ReasonCodes(UNSUBACK >> 4, identifier=c))
-            if len(reasoncodes) == 1:
-                reasoncodes = reasoncodes[0]
+                reasoncodes_list.append(ReasonCodes(UNSUBACK >> 4, identifier=c))
+
+            reasoncodes: typing.Union[
+                ReasonCodes, typing.List[ReasonCodes]
+            ] = reasoncodes_list
+            if len(reasoncodes_list) == 1:
+                reasoncodes = reasoncodes_list[0]
```

Comment:
Maybe add a TODO comment to remind someone to clean this up? It doesn't look very nice now. (I think `ReasonCodes` and `PacketTypes` should be named in the singular, for one...)

Reply:
The issue is that changing this will be a breaking change since it's used in on_subscribe callback.